test/e2e/tests/eigen_test.go

```diff
@@ -0,0 +1,560 @@
+	privateKey, err := crypto.GenerateKey()
+	require.NoError(t, err)
+
+	publicKey := privateKey.Public()
+	publicKeyECDSA, ok := publicKey.(*ecdsa.PublicKey)
+	require.Equal(t, true, ok)
+	address := crypto.PubkeyToAddress(*publicKeyECDSA)
+
+	// fund the account with ETH
+	transferFundTo(t, ctx, address.String(), client)
+
+	return address.String(), address, privateKey
+}
+
+func transferFundTo(t *testing.T, ctx context.Context, addr string, client *ethclient.Client) {
+	pk := mustHexToKey("0x4bbbf85ce3377467afe5d46f804f221813b2bb87f24d81f60f1fcdbf7cbf4356")
+	adr := common.HexToAddress("0x14dC79964da2C08b23698B3D3cc7Ca32193d9955")
+
+	nonce, err := client.PendingNonceAt(context.Background(), adr)
+	require.NoError(t, err)
+	value := big.NewInt(1000000000000000000) // 10 ETH
+	gasLimit := uint64(21000)
+	tip := big.NewInt(2000000000)
+	feeCap := big.NewInt(20000000000)
+	require.NoError(t, err)
+
+	var data []byte
+	chainID, err := client.NetworkID(context.Background())
+	require.NoError(t, err)
+
+	toAddress := common.HexToAddress(addr)
+	tx := ethtypes.NewTx(&ethtypes.DynamicFeeTx{
+		ChainID:   chainID,
+		Nonce:     nonce,
+		GasFeeCap: feeCap,
+		GasTipCap: tip,
+		Gas:       gasLimit,
+		To:        &toAddress,
+		Value:     value,
+		Data:      data,
+	})
+
+	signedTx, err := ethtypes.SignTx(tx, ethtypes.LatestSignerForChainID(chainID), pk)
+	require.NoError(t, err)
+
+	err = client.SendTransaction(context.Background(), signedTx)
+	require.NoError(t, err)
+	_, err = bind.WaitMined(ctx, client, signedTx)
+	require.NoError(t, err)
+}
+
+func checkIfContractsAreDeployed(
+	t *testing.T,
+	ctx context.Context,
+	avs AVS,
+	deployInfo map[types.ContractName]types.DeployInfo) {
+	t.Helper()
+
+	blockNumber, err := avs.Client.BlockNumber(ctx)
+	require.NoError(t, err)
+
+	// check if the contract has code in its respective contract addresses
+	checkIfCodePresent(t, ctx, avs, deployInfo, blockNumber, types.ContractOmniAVS)
+	checkIfCodePresent(t, ctx, avs, deployInfo, blockNumber, types.ContractELDelegationManager)
+	checkIfCodePresent(t, ctx, avs, deployInfo, blockNumber, types.ContractELAVSDirectory)
+	checkIfCodePresent(t, ctx, avs, deployInfo, blockNumber, types.ContractELStrategyManager)
+	checkIfCodePresent(t, ctx, avs, deployInfo, blockNumber, types.ContractELWETHStrategy)
+	checkIfCodePresent(t, ctx, avs, deployInfo, blockNumber, types.ContractELWETH)
+	checkIfCodePresent(t, ctx, avs, deployInfo, blockNumber, types.ContractELPodManager)
+}
+
+func checkOmniAVSConfigurations(
+	t *testing.T,
+	ctx context.Context,
+	avs AVS) {
+	t.Helper()
+
+	txOpts, err := bind.NewKeyedTransactorWithChainID(omniDepPk, big.NewInt(int64(avs.Chain.ID)))
+	require.NoError(t, err)
+	txOpts.Context = ctx
+
+	// set min stake
+	minStake := big.NewInt(MinStateForOperatorInEigenLayer)
+	tx, err := avs.AVSContract.SetMinimumOperatorStake(txOpts, minStake)
```

Comment:
Since this is a `check` function, it should probably not mutate state. Since we're now running transactions in tests we should probably structure these tests so that there is a clear: "given" (what is current state), "when" (action occurs), "then" ( what we expect)

Reply:
e2e tests will have mutations and checks... We should probably get used to this...
It's required to execute a flow from start to end.
As with the name of the functions, I can change it to execute and check....instead of check...